crates/uv/src/commands/tool/run.rs

```diff
@@ -118,9 +121,52 @@ pub(crate) async fn run(
         command.to_string_lossy(),
         args.iter().map(|arg| arg.to_string_lossy()).join(" ")
     );
-    let mut handle = process
-        .spawn()
-        .with_context(|| format!("Failed to spawn: `{}`", command.to_string_lossy()))?;
+    let mut handle = match process.spawn() {
+        Ok(handle) => Ok(handle),
+        Err(err) if err.kind() == std::io::ErrorKind::NotFound => {
+            match get_entrypoints(&from, &environment) {
+                Ok(entrypoints) => {
+                    if entrypoints.is_empty() {
+                        writeln!(
+                            printer.stdout(),
+                            "The executable {} was not found.",
+                            command.to_string_lossy().red(),
+                        )?;
+                    } else {
+                        writeln!(
+                            printer.stdout(),
+                            "The executable {} was not found.",
+                            command.to_string_lossy().red()
+                        )?;
+                        if has_from {
+                            writeln!(
+                                printer.stdout(),
+                                "However, the following executables are available:",
```

Comment:
Are executables from sub-depencencies considered as *available*? For instance, `uv tool run --from fastapi`, `uvicorn` is actually an available executable.

Reply:
I think we'll want a separate flag to opt-in to that, per https://github.com/astral-sh/uv/issues/4994

Maybe we should even warn or error if someone uses an executable that's not provided by the `--from` package?